<launch>

  # <arg name="server" default="localhost"/>

  <rosparam param="using_keyboard">False</rosparam>

  <param name="agent_num" type="int" value="2" />

  <rosparam param="car_ips">['172.16.0.207', '172.16.0.176']</rosparam>

  <rosparam param="car_targets">[[3.91, 1.05], [1.71, 3.95]]</rosparam>

  <node pkg="vrpn_client_ros" type="vrpn_client_node" name="vrpn_client_node" output="screen">
    <rosparam subst_value="true">
      # server: $(arg server)
      server: 172.16.0.7
      port: 3883

      update_frequency: 100.0
      frame_id: world

      # Use the VRPN server's time, or the client's ROS time.
      use_server_time: false
      broadcast_tf: true

      # Must either specify refresh frequency > 0.0, or a list of trackers to create
      refresh_tracker_frequency: 1.0
      #trackers:
      #- FirstTracker
      #- SecondTracker
    </rosparam>
  </node>

  <node pkg="vrpn_keyboard_control" type="vrpn_pose.py" name="vrpn_pose" output="screen">

  </node>

  <node pkg="vrpn_keyboard_control" type="rl_onserver.py" name="rl_onserver" output="screen">

    <param name="LASER_RANGE" type="double" value="1.4" />

    <param name="ROBOT_LENGTH" type="double" value="0.25" />

  </node>

  <node pkg="vrpn_keyboard_control" type="tcp_server.py" name="tcp_server" output="screen">

    <param name="tcp_ip" value="172.16.0.15" />

    <rosparam param="tcp_port">8888</rosparam>
  
  </node>
  
</launch>